types/src/extended_header.rs

```diff
@@ -61,14 +65,76 @@ const VERIFY_CLOCK_DRIFT: Duration = Duration::from_secs(10);
 /// genesis_header.verify(&fetched_header).expect("Malicious header received");
 /// ```
 #[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg(not(all(feature = "wasm-bindgen", target_arch = "wasm32")))]
 pub struct ExtendedHeader {
     /// Tendermint block header.
+    #[cfg_attr(
+        all(feature = "wasm-bindgen", target_arch = "wasm32"),
+        wasm_bindgen(skip)
+    )]
     pub header: Header,
     /// Commit metadata and signatures from validators committing the block.
+    #[cfg_attr(
+        all(feature = "wasm-bindgen", target_arch = "wasm32"),
+        wasm_bindgen(skip)
+    )]
     pub commit: Commit,
     /// Information about the set of validators commiting the block.
+    #[cfg_attr(
+        all(feature = "wasm-bindgen", target_arch = "wasm32"),
+        wasm_bindgen(skip)
+    )]
     pub validator_set: ValidatorSet,
     /// Header of the block data availability.
+    #[cfg_attr(
+        all(feature = "wasm-bindgen", target_arch = "wasm32"),
+        wasm_bindgen(getter_with_clone)
+    )]
+    pub dah: DataAvailabilityHeader,
+}
+
+/// Block header together with the relevant Data Availability metadata.
+///
+/// [`ExtendedHeader`]s are used to announce and describe the blocks
+/// in the Celestia network.
+///
+/// Before being used, each header should be validated and verified with a header you trust.
+///
+/// # Example
+///
+/// ```
+/// # use celestia_types::ExtendedHeader;
+/// # fn trusted_genesis_header() -> ExtendedHeader {
+/// #     let s = include_str!("../test_data/chain1/extended_header_block_1.json");
+/// #     serde_json::from_str(s).unwrap()
+/// # }
+/// # fn some_untrusted_header() -> ExtendedHeader {
+/// #     let s = include_str!("../test_data/chain1/extended_header_block_27.json");
+/// #     serde_json::from_str(s).unwrap()
+/// # }
+/// let genesis_header = trusted_genesis_header();
+///
+/// // fetch new header
+/// let fetched_header = some_untrusted_header();
+///
+/// fetched_header.validate().expect("Invalid block header");
+/// genesis_header.verify(&fetched_header).expect("Malicious header received");
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg(all(feature = "wasm-bindgen", target_arch = "wasm32"))]
+#[wasm_bindgen]
```

Comment:
I think the main point of making it wasm bindgenable was to also expose some methods on it, also getters for jsoned fields. without that, this looks less usable than just a json we had before